pkg/apis/controller/suggestions/v1beta1/suggestion_types.go

```diff
@@ -78,12 +78,15 @@ type TrialAssignment struct {
 	// Suggestion results with Trial parameters
 	ParameterAssignments []common.ParameterAssignment `json:"parameterAssignments,omitempty"`
 
-	//Name of the suggestion
+	// Name of the suggestion
 	Name string `json:"name,omitempty"`
 
 	// Rules for early stopping techniques
 	// Contains rule name, value and comparison type
 	EarlyStoppingRules []common.EarlyStoppingRule `json:"earlyStoppingRules,omitempty"`
+
+	// Suggestion metadata to attach to Trial job
+	Annotations map[string]string `json:"annotations,omitempty"`
```

Comment:
That makes sense @andreyvelich. 
In this use-case, it might be better to store it in annotions.

Reply:
For example, since BO algorithm doesn't generate Trial name, we have to search for HyperParameters based on Trial assignments: https://github.com/kubeflow/katib/blob/master/pkg/suggestion/v1beta1/skopt/base_service.py#L84-L98.

If Suggestion can generate Trial names, we can store these names in `recorded_trials_names` once SKOpt service [suggests new HPs](https://github.com/kubeflow/katib/blob/master/pkg/suggestion/v1beta1/skopt/base_service.py#L121).
Once `GetSuggestions` called again, we can easily update loss value in `recorded_trials_names` based on Trial name.